### ISSUE TITLE
Pyright error on correct code using class alias

### ISSUE THREAD

**Issue by aripollak** [microsoft/pyright#6002]
The following code causes pyright to throw an error:
```python
from astroquery.jplhorizons import Horizons

h = Horizons()
h.ephemerides_async
```

```
  /home/ari/foo.py:4:3 - error: Cannot access member "ephemerides_async" for type "BaseQuery"
    Member "ephemerides_async" is unknown (reportGeneralTypeIssues)
```

Changing `Horizons` to `HorizonsClass` makes pyright work as expected, with no error.

This uses [astroquery](https://pypi.org/project/astroquery/) 0.4.6, which aliases a HorizonsClass, which inherits from BaseQuery, which uses a metaclass, to Horizons: https://github.com/astropy/astroquery/blob/v0.4.6/astroquery/jplhorizons/core.py#L1308

I tried and failed to reproduce this with a smaller test case rather than requiring all of `astroquery`, so there must be something odd going on with the chain of classes/metaclasses that I'm not seeing. 

**Code or Screenshots**

**VS Code extension or command-line**
For the purposes of this test, I used pyright CLI 1.1.327.

**Comment by erictraut** [comment]
The symbol `Horizons` isn't a type alias. It's a variable whose type is inferred to be `HorizonsClass`. 

The `HorizonsClass` class doesn't provide a `__call__` method. Its superclass `BaseQuery` does provide a `__call__` method, but it contains no type annotations, so its return type must be inferred based on the `return` statements within the method. 

```python
    def __call__(self, *args, **kwargs):
        """ init a fresh copy of self """
        return self.__class__(*args, **kwargs)
```

The inferred return type is `BaseQuery`, which is what pyright is evaluating for the value of `Horizons()`.

So it looks like pyright is working correctly here.

Your options:
1. Work with the maintainers of `astroquery` to add type annotations to the library.
2. Write your own type stubs for the `astroquery` library (or at least the portions that you're using).
3. Set `useLibraryCodeForTypes` to false within the pyrightconfig.json configuration file. This tells pyright to assume that all symbols imported from untyped libraries are of type `Unknown`.
4. Use an explicit `cast` within your code, as follows:
```python
from typing import cast
from astroquery.jplhorizons import Horizons, HorizonsClass

h = cast(HorizonsClass, Horizons())
```
